src/Common/Config/ConfigProcessor.cpp

```diff
@@ -673,7 +673,9 @@ XMLDocumentPtr ConfigProcessor::processConfig(
     zkutil::ZooKeeperNodeCache * zk_node_cache,
     const zkutil::EventPtr & zk_changed_event)
 {
-    LOG_DEBUG(log, "Processing configuration file '{}'.", path);
+    const bool write_logs = is_config_changed;
```

Comment:
maybe we could not call this function at all if `is_config_changed == false`?

Reply:
Just tried it, it seems not really trivial, maybe revert last changes?

Reply:
it seems we already have a check for `mtime` few lines above: https://github.com/ClickHouse/ClickHouse/blob/0dc01f4a949ccf6f2db8c2dd27431fee879849cd/src/Common/Config/ConfigReloader.cpp#L114
so it seems we already shouldn't update config if files didn't change

Reply:
But two other conditions don't require files to change (and to write logs about a new configuration accordingly)

Reply:
ok, then first patch makes sense. but i'm still not sure how often `force` or `need_reload_from_zk` can be true.